exporter/clickhouseexporter/exporter_logs.go

```diff
@@ -133,34 +133,37 @@ const (
-     SpanId String CODEC(ZSTD(1)),
-     TraceFlags UInt32 CODEC(ZSTD(1)),
-     SeverityText LowCardinality(String) CODEC(ZSTD(1)),
-     SeverityNumber Int32 CODEC(ZSTD(1)),
-     ServiceName LowCardinality(String) CODEC(ZSTD(1)),
-     Body String CODEC(ZSTD(1)),
-     ResourceSchemaUrl String CODEC(ZSTD(1)),
-     ResourceAttributes Map(LowCardinality(String), String) CODEC(ZSTD(1)),
-     ScopeSchemaUrl String CODEC(ZSTD(1)),
-     ScopeName String CODEC(ZSTD(1)),
-     ScopeVersion String CODEC(ZSTD(1)),
-     ScopeAttributes Map(LowCardinality(String), String) CODEC(ZSTD(1)),
-     LogAttributes Map(LowCardinality(String), String) CODEC(ZSTD(1)),
-     INDEX idx_trace_id TraceId TYPE bloom_filter(0.001) GRANULARITY 1,
-     INDEX idx_res_attr_key mapKeys(ResourceAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
-     INDEX idx_res_attr_value mapValues(ResourceAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
-     INDEX idx_scope_attr_key mapKeys(ScopeAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
-     INDEX idx_scope_attr_value mapValues(ScopeAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
-     INDEX idx_log_attr_key mapKeys(LogAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
-     INDEX idx_log_attr_value mapValues(LogAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
-     INDEX idx_body Body TYPE tokenbf_v1(32768, 3, 0) GRANULARITY 1
+	Timestamp DateTime64(9) CODEC(Delta(8), ZSTD(1)),
+	TimestampDate Date DEFAULT toDate(Timestamp),
+	TimestampTime DateTime DEFAULT toDateTime(Timestamp),
+	TraceId String CODEC(ZSTD(1)),
+	SpanId String CODEC(ZSTD(1)),
+	TraceFlags UInt8,
+	SeverityText LowCardinality(String) CODEC(ZSTD(1)),
+	SeverityNumber UInt8,
+	ServiceName LowCardinality(String) CODEC(ZSTD(1)),
+	Body String CODEC(ZSTD(1)),
+	ResourceSchemaUrl LowCardinality(String) CODEC(ZSTD(1)),
+	ResourceAttributes Map(LowCardinality(String), String) CODEC(ZSTD(1)),
+	ScopeSchemaUrl LowCardinality(String) CODEC(ZSTD(1)),
+	ScopeName String CODEC(ZSTD(1)),
+	ScopeVersion LowCardinality(String) CODEC(ZSTD(1)),
+	ScopeAttributes Map(LowCardinality(String), String) CODEC(ZSTD(1)),
+	LogAttributes Map(LowCardinality(String), String) CODEC(ZSTD(1)),
+
+	INDEX idx_trace_id TraceId TYPE bloom_filter(0.001) GRANULARITY 1,
+	INDEX idx_res_attr_key mapKeys(ResourceAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+	INDEX idx_res_attr_value mapValues(ResourceAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+	INDEX idx_scope_attr_key mapKeys(ScopeAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+	INDEX idx_scope_attr_value mapValues(ScopeAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+	INDEX idx_log_attr_key mapKeys(LogAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+	INDEX idx_log_attr_value mapValues(LogAttributes) TYPE bloom_filter(0.01) GRANULARITY 1,
+	INDEX idx_body Body TYPE tokenbf_v1(32768, 3, 0) GRANULARITY 1
 ) ENGINE = %s
+PARTITION BY toYYYYMM(TimestampDate)
+ORDER BY (TimestampDate, TimestampTime)
```

Comment:
general LGTM.  just wondering `ORDER BY (TimestampDate, TimestampTime)` may not better than ORDER BY ServiceName, SeverityText, toUnixTimestamp(Timestamp), beacause [ServiceName](https://opentelemetry.io/docs/specs/semconv/resource/#service) and SeverityText( also as well-known logLevel) is the most common filter when query service log, can you write more text explain why remove them? thanks.

Reply:
It can be difficult to model a decent table that works for everyone (consider issues like https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32215)

The idea behind this change is to simply make time range queries work as best they can. Every log has a timestamp, but not always a proper `ServiceName` or `Severity*`. Some OTel data isn't composed very well. `SeverityText` might not be as good as `SeverityNumber`, and if we try to add all of these to `ORDER BY` then it starts to lose its purpose as a primary index.

For those who will be using this exporter in production, the goal is to have them use this as a starting point, and then add the appropriate columns to the `ORDER BY` depending on their primary column. For example, look at how [this blog post](https://clickhouse.com/blog/building-a-logging-platform-with-clickhouse-and-saving-millions-over-datadog#schema) chooses to use `PodName` and `Timestamp`:
```sql
ORDER BY (PodName, Timestamp)
```

What we've found is that a lot of these values are not necessary in the order by, especially ones after the `Timestamp*` columns (such as `TraceId` in the current version). Most log queries are going to be in a tight range of time (5, 15, 30, 60 minutes) and these smaller `ORDER BY` clauses favor this. Again, you can always go a step further by adding your preferred column to the start of it, such as `ServiceName`, `SeverityText`, `PodName`, or some combination of these.

The goal is to have a good default that is easy for others to expand on for their own deployment, while also still giving a good default for those who *don't* bother to configure one. If you're at the scale where this schema becomes a problem, you would already be using your own schema.

Reply:
ServiceName is a required in spec.  https://opentelemetry.io/docs/specs/semconv/resource/#service. to say the least, ServiceName is needed. see also as https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31670

in `PodName` blog case , i think most common usage query pattern is `PodName like xxx%`, not `PodName= xxx`, mostly we query a workload log, not the specific pod log. actually, use a serviceName as workload name will be more reasonaly.

Reply:
This is reasonable enough since it is a default schema, and if someone needs to change it they're free to do so for their deployment.

As you suggested, I have added `ServiceName` to the default schema.